src/main/java/de/tum/cit/ase/ares/api/util/FileTools.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    //<editor-fold desc="Copy">
+    /**
+     * Copies a list of files to a specified target directory.
+     * <p>
+     * This method iterates over the provided list of source file paths, copying each file
+     * to the specified target directory. It handles various exceptions that might occur
+     * during the copy process, wrapping them in a {@link SecurityException}.
+     * </p>
+     *
+     * @param sourceFilePaths a {@link List} of {@link Path} objects representing the source files to copy.
+     * @param targetFilePaths a {@link List} of {@link Path} objects representing the target paths for the copied files.
+     * @return a {@link List} of {@link Path} objects representing the paths of the copied files.
+     * @throws SecurityException if an error occurs during the file copy process.
+     */
+    public static List<Path> copyFiles(List<Path> sourceFilePaths, List<Path> targetFilePaths) {
+        return IntStream.range(0, sourceFilePaths.size()).mapToObj(i -> {
+            try (InputStream sourceStream = FileTools.class.getResourceAsStream("/" + sourceFilePaths.get(i).toString())) {
+                if (sourceStream == null) {
+                    throw new IOException("Resource not found: " + sourceFilePaths.get(i));
+                }
+                if (!Files.exists(targetFilePaths.get(i).getParent())) {
+                    Files.createDirectories(targetFilePaths.get(i).getParent());
+                }
+                Files.copy(sourceStream, targetFilePaths.get(i), StandardCopyOption.REPLACE_EXISTING);
+            } catch (InvalidPathException e) {
+                throw new SecurityException("Ares Security Error (Stage: Creation): Invalid path provided during file copy.", e);
+            } catch (UnsupportedOperationException e) {
+                throw new SecurityException("Ares Security Error (Stage: Creation): Unsupported operation during file copy.", e);
+            } catch (FileAlreadyExistsException e) {
+                throw new SecurityException("Ares Security Error (Stage: Creation): File already exists at target location.", e);
+            } catch (IOException e) {
+                throw new SecurityException("Ares Security Error (Stage: Creation): IO error occurred during file copy.", e);
+            }
+            return targetFilePaths.get(i);
+        }).toList();
+    }
+
+    public static List<Path> copyJavaFiles(List<Path> sourceFilePaths, List<Path> targetFilePaths, List<String[]> formatValues) {
+        List<Path> copiedFiles = copyFiles(sourceFilePaths, targetFilePaths);
+        for (int i = 0; i < copiedFiles.size(); i++) {
+            try {
+                String formatedFile;
+                try{
+                    formatedFile = String.format(Files.readString(copiedFiles.get(i)), (String[]) formatValues.get(i));
+                } catch (IllegalFormatException e) {
+                    System.out.println("Illegal format in " + copiedFiles.get(i).toAbsolutePath());
```

Comment:
Should this be a System.out.println or also a SecurityException?

Reply:
Addressed in [e5a7f84a664b8e666a8cd00b5e15224a6946da2d](https://github.com/ls1intum/Ares2/pull/25/commits/e5a7f84a664b8e666a8cd00b5e15224a6946da2d)